### PR TITLE
[FEMDEM] Correcting the compilation in Win & cleaning

### PR DESCRIPTION
**Pull request by AlejandroCornejo**
by the way, @AFranci I commented a line in Solid MEch App that was not thread safe (crashes my calculation)

### CODE REVIEW

File: applications/FemToDemApplication/custom_constitutive/elastic_isotropic_3d.cpp [hunk 1]
```diff
@@ -17,6 +17,7 @@
 #include "custom_constitutive/elastic_isotropic_3d.h"
 #include "includes/checks.h"
 #include "fem_to_dem_application_variables.h"
+#include "solid_mechanics_application_variables.h"
```
Comment (loumalouomega) [inline applications/FemToDemApplication/custom_constitutive/elastic_isotropic_3d.cpp]: This is not the best way to reduce dependencies. If you just need the variables, use the KratosComponents

**Review by loumalouomega — CHANGES_REQUESTED** [review]
Use KratosComponents instead of including new (and temporal) dependencies

File: applications/FemToDemApplication/custom_constitutive/elastic_isotropic_3d.cpp (hunk 1 above)
Comment (AlejandroCornejo) [reply applications/FemToDemApplication/custom_constitutive/elastic_isotropic_3d.cpp]: Do you have an example??? I'd be grateful :)

Comment (loumalouomega) [reply applications/FemToDemApplication/custom_constitutive/elastic_isotropic_3d.cpp]:
~~~cpp
const auto& r_variable = KratosComponents<Variable<double>>::Get("RAYLEIGH_ALPHA");
~~~

Can be used to any variable, for components an adaptor is needed. Check the custom_bdf scheme in the core

Comment (AlejandroCornejo) [reply applications/FemToDemApplication/custom_constitutive/elastic_isotropic_3d.cpp]: I've tried and I got a Seg fault... :S

Comment (loumalouomega) [reply applications/FemToDemApplication/custom_constitutive/elastic_isotropic_3d.cpp]: Strange, did you put the &?

Comment (AlejandroCornejo) [reply applications/FemToDemApplication/custom_constitutive/elastic_isotropic_3d.cpp]: nevermind, solved hehehe

**Comment by AlejandroCornejo** [comment]
@loumalouomega I've done the changes, approve whenever you're ready :) thank you

**Review by loumalouomega — APPROVED** [review]
OK

**Comment by loumalouomega** [comment]
BTW, maybe is a good idea to change back the variables once you move to the SMA. In the place where you are using it is quite expensive to use the KratosComponents

**Pull request merged**